omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptStateConversionController2_0Test.java

```diff
@@ -0,0 +1,124 @@
+import org.openmrs.Concept;
+import org.openmrs.ConceptStateConversion;
+import org.openmrs.ProgramWorkflow;
+import org.openmrs.ProgramWorkflowState;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.RestTestConstants1_8;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+public class ConceptStateConversionController2_0Test extends MainResourceControllerTest {
+
+	private static String uuid;
+
+	@Before
+	public void setUp() {
+		Concept concept = Context.getConceptService().getConceptByUuid(RestTestConstants1_8.CONCEPT2_UUID);
+		ProgramWorkflow workflow = Context.getProgramWorkflowService().getWorkflowByUuid(RestTestConstants1_8.WORKFLOW_UUID);
+
+		ProgramWorkflowState state = new ProgramWorkflowState();
+		state.setConcept(concept);
+		state.setInitial(true);
+		state.setTerminal(false);
+
+		workflow.addState(state);
+		Context.getProgramWorkflowService().saveProgram(workflow.getProgram());
+
+		Context.flushSession();
+
+		ConceptStateConversion conceptStateConversion = new ConceptStateConversion();
+		conceptStateConversion.setConcept(concept);
+		conceptStateConversion.setProgramWorkflow(workflow);
+		conceptStateConversion.setProgramWorkflowState(state);
+
+		Context.getProgramWorkflowService().saveConceptStateConversion(conceptStateConversion);
+		uuid = conceptStateConversion.getUuid();
+	}
+
+	@Override
+	public String getURI() {
+		return "conceptstateconversion";
+	}
+
+	@Override
+	public String getUuid() {
+		return uuid;
+	}
+
+	@Override
+	public long getAllCount() {
+		return 2;
+	}
+
+	@Test
+	public void shouldCreateStateConversion() throws Exception {
+		Concept concept = Context.getConceptService().getConceptByUuid("0955b484-b364-43dd-909b-1fa3655eaad2");
+		ProgramWorkflow workflow = Context.getProgramWorkflowService().getWorkflowByUuid(RestTestConstants1_8.WORKFLOW_UUID);
+
+		ProgramWorkflowState state = new ProgramWorkflowState();
+		state.setConcept(concept);
+		state.setInitial(true);
+		state.setTerminal(false);
+
+		workflow.addState(state);
+		Context.getProgramWorkflowService().saveProgram(workflow.getProgram());
+
+		Context.flushSession();
+
+		String json =
+				"{\"concept\": \"" + concept.getUuid() + "\",\"programWorkflow\": \"" + workflow.getUuid()
+						+ "\",\"programWorkflowState\": \"" + state.getUuid() + "\"}";
+
+		SimpleObject newStateConversion = deserialize(handle(newPostRequest(getURI(), json)));
+
+		assertNotNull(newStateConversion);
+		String uuid = newStateConversion.get("uuid");
+
+		ConceptStateConversion createdStateConversion = Context.getProgramWorkflowService().getConceptStateConversionByUuid(uuid);
+		assertEquals(workflow.getUuid(), createdStateConversion.getProgramWorkflow().getUuid());
```

Comment:
Can you also assert that the total count has increased?

Reply:
https://github.com/openmrs/openmrs-module-webservices.rest/pull/493/commits/a2fa31aba58d4a72235cd93c5293e50b2c05e0f4